app/controllers/campaign_pages_controller.rb

```diff
@@ -15,8 +15,9 @@ def new
 
   def create
     @campaign_page = CampaignPage.new(@page_params)
-    @campaign_page.compile_html
+    return switch_template(@campaign_page, :new) if params[:switch_template]
```

Comment:
You could make this more DRY. Why not have all of this detail handled by another class? Right now, this controller is doing too much: "Provide CRUD actions for campaign pages **AND** deal with situations where client wants to switch templates." Can we ditch the second clause?